syntax = "proto3";

package svg_proto;

message Point {
  double x = 1;
  double y = 2;
}

message RGB {
  uint32 red = 1;
  uint32 green = 2;
  uint32 blue = 3;
}

message RGBA {
  uint32 red = 1;
  uint32 green = 2;
  uint32 blue = 3;
  double opacity = 4;
}

message Color {
  oneof variant {
    bytes string_color = 1;
    RGB rgb_color = 2;
    RGBA rgba_color = 3;
  }
}
